src/Core/Misc/ExtensionMethods.cs

```diff
@@ -647,9 +647,11 @@ public static string GetUIFramework(this IA11yElement e)
         {
             if (e == null) return null;
 
-            return (string.IsNullOrEmpty(e.Framework))
+            string framework = e.Framework;
+
+            return (string.IsNullOrEmpty(framework))
```

Comment:
While running the unit tests, I noticed that e.Framework was being called twice. I had the choice to tell the unit tests to double their expected number of calls, or change the Extension method to only query the property once. The latter seemed like the better choice. Happy to discuss if you wish.

Reply:
I think this needs a comment in the code.

Reply:
Comment added